src/Tasks/ManifestUtil/Manifest.cs

```diff
@@ -485,6 +498,21 @@ private static void UpdateFileReference(BaseReference f, string targetFrameworkV
             }
             f.Hash = hash;
             f.Size = size;
+
+            //
+            // .NETCore Launcher.exe based Deployment: If the filereference is for apphost.exe, we need to change
+            // the ResolvedPath and TargetPath to {assemblyname}.exe before we write the manifest, so that the
+            // manifest does not have a file reference to apphost.exe
+            //
+            string fileName = Path.GetFileName(f.ResolvedPath);
+            if (LauncherBasedDeployment &&
+                fileName.Equals(Constants.AppHostExe, StringComparison.InvariantCultureIgnoreCase) &&
+                !String.IsNullOrEmpty(AssemblyName))
+            {
+                f.ResolvedPath = Path.Combine(Path.GetDirectoryName(f.ResolvedPath), AssemblyName);
```

Comment:
Does `AssemblyName` have an extension? Based on the property you're passing in would `ApplicationManifestIdentity` be a better name?

Reply:
We use AssemblyName already in an existing task (GenerateApplicationManifest) so I chose to keep it the same for consistency.